lib/consul/index.js

```diff
@@ -198,8 +247,18 @@ var file_modified = function(branch, file, cb) {
   var fqf = branch.branch_directory + path.sep + file;
   logger.trace('Attempting to read "%s"', fqf);
 
-  if (branch.expand_keys) {
-    handle_expanded_keys_with_different_file_types();
+  if (branch.expand_keys && branch.expand_keys_diff) {
+    get_kvs(branch, file, function(err, kvs) {
+      if (err) return cb('Failed to load KV tree ' + create_key_name(branch, file) + ' due to ' + err);
+      handle_expanded_keys_with_different_file_types(kvs);
+    });
+  } else if (branch.expand_keys) {
+    // Delete current tree.  Yeah, I know this is kinda the coward's way out, but it's a hell of a lot
```

Comment:
Edit/remove this comment, since you have the option to _not_ do this any more? After all, your was goal to get rid of this comment, right? :)